internal/packagejson/cache.go

```diff
@@ -121,9 +122,8 @@ func (p *InfoCacheEntry) GetDirectory() string {
 }
 
 type InfoCache struct {
-	mu                        sync.RWMutex
-	IsReadonly                bool
-	cache                     map[tspath.Path]InfoCacheEntry
+	cache                     collections.SyncMap[tspath.Path, *InfoCacheEntry]
+	isReadonly                atomic.Bool
```

Comment:
Interestingly, this field is never set to `true`, and that's been the case since #79. Probably, it could just be removed.

Reply:
For reference, in Strada, this is set in `startCachingPerDirectoryResolution` and `finishCachingPerDirectoryResolution`. I believe we dropped a lot of this in #796, but maybe that's unrelated.

Reply:
I think @andrewbranch or @sheetalkamat would know better. Certainly the model where we have one cache and mutate it will not fly in this codebase, so I suspect we would be safe to drop it.